client/src/chat/components/hooks/useOutsideClick.tsx

```diff
@@ -0,0 +1,21 @@
+import { useRef, useState, useEffect } from "react";
+
+const useOutsideClick = (initialValue: boolean) => {
```

Comment:
Naming here is a bit weird; the name 'useOutsideClick' doesn't really impart what this function is doing. It also uses names like 'showEmojiPicker' internally, but the function doesn't necessarily have anything to do with emoji